src/worker.cpp

```diff
@@ -1242,7 +1242,11 @@ void Worker::collectFrusFromJson()
         const std::string& vpdFilePath = itemFRUS.key();
 
         // skip processing of system VPD again as it has been already collected.
-        if (vpdFilePath == SYSTEM_VPD_FILE_PATH)
+        // Also, if chassis is powered on, skip collecting FRUs which are
+        // powerOffOnly.
+        if (vpdFilePath == SYSTEM_VPD_FILE_PATH ||
+            (dbusUtility::isChassisPowerOn() &&
```

Comment:
Can we switch the LHS and RHS of && condition? Don;t want to make unnecessary Dbus calls for FRUs which actually does not need this info.

Reply:
Addressed.